examples/ogbn_train_cugraph.py

```diff
@@ -0,0 +1,293 @@
+
+import torch_geometric  # noqa
+from torch_geometric.utils import to_undirected  # noqa
+
+
+def arg_parse():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter, )
+    parser.add_argument(
+        '--dataset',
+        type=str,
+        default='ogbn-papers100M',
+        choices=['ogbn-papers100M', 'ogbn-products', 'ogbn-arxiv'],
+        help='Dataset name.',
+    )
+    parser.add_argument(
+        '--dataset_dir',
+        type=str,
+        default='/workspace/data',
+        help='Root directory of dataset.',
+    )
+    parser.add_argument(
+        "--dataset_subdir",
+        type=str,
+        default="ogb-papers100M",
+        help="directory of dataset.",
+    )
+    parser.add_argument('-e', '--epochs', type=int, default=10)
+    parser.add_argument('--num_layers', type=int, default=3)
+    parser.add_argument('-b', '--batch_size', type=int, default=1024)
+    parser.add_argument('--fan_out', type=int, default=10)
+    parser.add_argument('--hidden_channels', type=int, default=256)
+    parser.add_argument('--lr', type=float, default=0.003)
+    parser.add_argument('--wd', type=float, default=0.0,
+                        help='weight decay for the optimizer')
+    parser.add_argument('--dropout', type=float, default=0.5)
+    parser.add_argument('--num_workers', type=int, default=12)
+    parser.add_argument(
+        '--use_directed_graph',
+        action='store_true',
+        help='Whether or not to use directed graph',
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        default='SAGE',
+        choices=['SAGE', 'GAT', 'GCN'],
+        help="Model used for training, default GraphSAGE",
+    )
+    parser.add_argument(
+        "--num_gat_conv_heads",
+        type=int,
+        default=4,
+        help="If using GATConv, number of attention heads to use",
+    )
+    parser.add_argument('--tempdir_root', type=str, default=None)
+    args = parser.parse_args()
+
+    return args
+
+
+def create_loader(
+    data,
+    num_neighbors,
+    input_nodes,
+    replace,
+    batch_size,
+    samples_dir,
+    stage_name,
+    shuffle=False,
+):
+    directory = osp.join(samples_dir, stage_name)
+    os.mkdir(directory)
+    return NeighborLoader(
+        data,
+        num_neighbors=num_neighbors,
+        input_nodes=input_nodes,
+        replace=replace,
+        batch_size=batch_size,
+        directory=directory,
+        shuffle=shuffle,
+    )
+
+
+def train(model, train_loader):
+    model.train()
+
+    total_loss = total_correct = total_examples = 0
+    for i, batch in enumerate(train_loader):
+        batch = batch.cuda()
+        optimizer.zero_grad()
+        out = model(batch.x, batch.edge_index)[:batch.batch_size]
+        y = batch.y[:batch.batch_size].view(-1).to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss) * y.size(0)
```

Comment:
Let's not call `float()` to avoid unnecessary D2H sync every step.

Reply:
total_loss += loss.item() * y.size(0)

Reply:
Let's not call `.item()` here. You can call it once at the end of a training epoch.
```suggestion
        total_loss += loss * y.size(0)
```